### DISCUSSION TITLE
`sanitizeCommitMessage` handlebars helper

### DISCUSSION THREAD

**Discussion by mmkal** [renovatebot/renovate#31448] [Suggest an Idea]
### Tell us more.

CC @rarkins, related to https://github.com/renovatebot/renovate/discussions/31230 - I would be happy to help with this if you could point me to the right area of code. I haven't contributed to renovate for a few years now.

Based on annoying pings that can be caused by putting release notes in `commitBody`.

If using handlebars, it looks like a [custom helper](https://github.com/renovatebot/renovate/discussions/31230) is what we're looking for. I think an initial implementation could be as simple as:

```
s => s.replaceAll('@', '`@`')
```

But it might be a good idea to escape links and so on too, since they can also cause noise (another example from my custom setup: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2346 - though worth noting that I still think my setup was pretty innocent!)

<img width="825" alt="image" src="https://github.com/user-attachments/assets/df95a818-e633-46de-84de-27e1a7e20635">

**Comment by viceice** [comment]
there's already a template helper for replace 

https://docs.renovatebot.com/templates/#replace

**Reply by mmkal** [reply]
Ah ok, so do you think something like this would do the trick?

```
{{{replace '(@|github.com)' '`$1`'  release.releaseNotes.body}}}
```

(the idea being, put surround that tend to trigger github notifications in backticks- I've assumed here that we can use `$1` to get the captured group)

I want to be a bit careful to avoid spamming people again. The full commitBody would then be:

```
  "commitBody": "{{#if logJSON.hasReleaseNotes}}{{#each logJSON.versions as |release|}}{{# if release.releaseNotes}}##### [`v{{{release.version}}}]({{{release.releaseNotes.url}}})\n\n{{{replace '(@|github.com)' '`$1`'  release.releaseNotes.body}}}{{/if}}{{/each}}{{/if}}"
```

If so, it's probably worth updating the example repo linked to [here](https://github.com/renovatebot/renovate/discussions/30727#discussioncomment-10319016).

**Reply by rarkins** [reply]
You can test from private repos - those won't trigger back links.
